src/core/project/qgsproject.h

```diff
@@ -370,13 +370,36 @@ class CORE_EXPORT QgsProject : public QObject, public QgsExpressionContextGenera
     /**
      * Returns the project's native coordinate reference system.
      *
+     * \warning Since QGIS 3.38, consider using crs3D() whenever transforming 3D data or whenever
+     * z/elevation value handling is important.
+     *
      * \see setCrs()
+     * \see crs3D()
      * \see verticalCrs()
      * \see ellipsoid()
      * \see crsChanged()
      */
     QgsCoordinateReferenceSystem crs() const;
 
+    /**
+     * Returns the CRS to use for the project when transforming 3D data, or when z/elevation
+     * value handling is important.
+     *
+     * The returned CRS will take into account verticalCrs(), e.g. by returning a compound
+     * CRS consisting of crs() + verticalCrs(). This method may still return a 2D CRS, e.g in the
+     * case that crs() is a 2D CRS and no verticalCrs() has been set for the project.
```

Comment:
Maybe it is worth noting how one can verify if the returned crs is actually 3d or 2d?

Reply:
Ok, done